credoai/modules/model_modules/nlp_generator.py

```diff
@@ -0,0 +1,372 @@
+            One of the following:
+                Name of a builtin prompt dataset. Choices are
+                    'bold_gender', 'bold_political_ideology', 'bold_profession', 'bold_race', 'bold_religious_ideology' (from Dhamala et al. 2021)
+                    'realtoxicityprompts_1000', 'realtoxicityprompts_challenging_20', 'realtoxicityprompts_challenging_100', 'realtoxicityprompts_challenging' (from Gehman et al. 2020)
+                Path of your own prompts csv file with columns 'group', 'subgroup', 'prompt'
+
+        Returns
+        -------
+        pandas.dataframe
+            prompts dataframe with columns
+            Schema: ['group', 'subgroup', 'prompt']
+
+        Raises
+        ------
+        Exception
+            Occurs if the provided prompts dataset csv file is not a valid file
+        Exception
+            Occurs if the prompts dataset cannot be loaded
+        """
+        if prompts in PROMPTS_PATHS:
+            prompts_path = get_data_path(PROMPTS_PATHS[prompts])
+            df = pd.read_csv(prompts_path)
+
+        elif prompts.split(".")[-1] == "csv":
+            df = pd.read_csv(prompts)
+            cols_required = ["group", "subgroup", "prompt"]
+            cols_given = list(df.columns)
+            if set(cols_given) != set(cols_required):
+                cols_required_str = ", ".join(cols_required)
+                raise Exception(
+                    "The provided prompts dataset csv file is not a valid file. Ensure it has all and only the following columns: "
+                    + cols_required_str
+                )
+
+        else:
+            builtin_prompts_names = list(PROMPTS_PATHS.keys())
+            builtin_prompts_names = ", ".join(builtin_prompts_names)
+            raise Exception(
+                "The prompts dataset cannot be loaded. Ensure the provided prompts value is either a path to a valid csv file"
+                + " or name of one of the builtin datasets (i.e."
+                + builtin_prompts_names
+                + "). You provided "
+                + prompts
+            )
+
+        return df
+
+    def _perform_prerun_checks(self):
+        """Checks the provided configurations and the generation and assessment functions
+
+        Raises
+        ------
+        Exception
+            Occurs if checks are not successfully completed
+        """
+        # Check types
+        for item in [self.generation_functions, self.assessment_functions]:
+            if not isinstance(item, dict):
+                raise Exception(
+                    "'generation_functions' and 'assessment_functions' values must be of type dict."
+                )
+
+        # Check the generation functions
+        test_prompt = "To be, or not to be, that is"
+        for gen_name, gen_fun in self.generation_functions.items():
+            try:
+                response = gen_fun(test_prompt)
+                if not isinstance(response, str):
+                    raise Exception(
```

Comment:
All of these exceptions should be made `ValidationErrors` (see credoai.utils.common). I am making some errors we can return that are more informative. This prerun_check is definitely a form of validation

Reply:
I see you have now made the updates accordingly. Awesome.